metadata/src/main/java/com/redhat/lightblue/metadata/FieldConstraint.java

```diff
@@ -31,4 +31,10 @@
      * Determines if the constraint is valid for the given field type
      */
     boolean isValidForFieldType(Type fieldType);
+
+
+    /**
+     * Returns the description of the FieldConstraint (for documentation propose)
+     */
+    String getDescription();
```

Comment:
Why does the constraint need a description?  Each property in the json schema needs a description, yes, but those are given by the field in lightblue metadata.  Constraints don't have a description in json schema.